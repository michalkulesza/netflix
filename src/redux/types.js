//error.js
export const SET_ERROR = "SET_ERROR";
export const CLEAR_ERROR = "CLEAR_ERROR";

//fetch-details.js
export const FETCH_DETAILS = "FETCH_DETAILS";
export const CLEAR_DETAILS = "CLEAR_DETAILS";

//fetch-episodes.js
export const FETCH_EPISODES = "FETCH_EPISODES";
export const SET_EPISODES_UPDATING = "SET_EPISODES_UPDATING";
export const CLEAR_EPISODES = "CLEAR_EPISODES";

//fetch-genre-data.js
export const FETCH_GENRE_DATA = "FETCH_GENRE_DATA";
export const FETCH_MORE_GENRE_DATA = "FETCH_MORE_GENRE_DATA";
export const SET_GENRE_DATA_UPDATING = "SET_GENRE_DATA_UPDATING";
export const CLEAR_GENRE_DATA = "CLEAR_GENRE_DATA";

//fetch-initial-data.js
export const SET_INITIAL_DATA = "SET_INITIAL_DATA";
export const SET_DATA_UPDATING = "SET_DATA_UPDATING";
export const CLEAR_INITIAL_DATA = "CLEAR_INITIAL_DATA";

//genres.js
export const SET_GENRES = "SET_GENRES";
export const CLEAR_GENRES = "CLEAR_GENRES";
export const SET_SELECTED_GENRE = "SET_SELECTED_GENRE";
export const CLEAR_SELECTED_GENRE = "CLEAR_SELECTED_GENRE";

//misc.js
export const SET_DETAILS_POSITION = "SET_DETAILS_POSITION";
export const SET_EXPANDED_POSITION = "SET_EXPANDED_POSITION";
export const CLEAR_EXPANDED_POSITION = "CLEAR_EXPANDED_POSITION";
export const SET_EXPANDED_TRANSFORM_ORIGIN = "SET_EXPANDED_TRANSFORM_ORIGIN";
export const SET_HEADER_VIDEO = "SET_HEADER_VIDEO";
export const SET_PRESSED_KEY = "SET_PRESSED_KEY";
export const SET_SCROLLBAR_WIDTH = "SET_SCROLLBAR_WIDTH";
export const SET_ACTIVE_EXPANDED = "SET_ACTIVE_EXPANDED";
export const CLEAR_ACTIVE_EXPANDED = "CLEAR_ACTIVE_EXPANDED";
export const CLEAR_MISC = "CLEAR_MISC";

//player.js
export const SET_PLAYER_TV = "SET_PLAYER_TV";
export const SET_PLAYER_FILM = "SET_PLAYER_FILM";
export const SET_PLAYER_VOLUME = "SET_PLAYER_VOLUME";
export const SET_PLAYER_SEASONS = "SET_PLAYER_SEASONS";
export const SET_PLAYER_STATE = "SET_PLAYER_STATE";
export const SET_PLAYER_META_LOADED = "SET_PLAYER_META_LOADED";
export const CLEAR_PLAYER = "CLEAR_PLAYER";

//toggles.js
export const SET_IS_DETAILS = "SET_IS_DETAILS";
export const SET_GLOBAL_MUTE = "SET_GLOBAL_MUTE";
export const CLEAR_TOGGLES = "CLEAR_TOGGLES";

//user.js
export const SET_USER_INFO = "SET_USER_INFO";
export const SET_USER = "SET_USER";
export const TOGGLE_LIKE_VIDEO = "TOGGLE_LIKE_VIDEO";
export const DISLIKE_VIDEO = "DISLIKE_VIDEO";
export const REMOVE_DISLIKED_VIDEO = "REMOVE_DISLIKED_VIDEO";
export const TOGGLE_VIDEO_LIST = "TOGGLE_VIDEO_LIST";
export const CLEAR_USER = "CLEAR_USER";
