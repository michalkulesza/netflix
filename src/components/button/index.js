import React from "react";
import { Main, Square, Round, Container, Label, Clear, Form } from "./styles/button";

import { ImSpinner8 } from "react-icons/im";

const Button = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Button.Square = ({ iconScale, inverted, marginRight, children, ...restProps }) => {
	return (
		<Square iconScale={iconScale} inverted={inverted} marginRight={marginRight} {...restProps}>
			{children}
		</Square>
	);
};

Button.Round = ({ label, inverted, dark, marginRight, children, isLoading, ...restProps }) => {
	return (
		<Container marginRight={marginRight}>
			<Round inverted={inverted} dark={dark} isLoading={isLoading} {...restProps}>
				{children}
			</Round>
			{label && <Label>{label}</Label>}
		</Container>
	);
};

Button.Clear = ({ size, padding, children, margin, ...restProps }) => {
	return (
		<Clear size={size} padding={padding} margin={margin} {...restProps}>
			{children}
		</Clear>
	);
};

Button.Form = ({ loading, children, ...restProps }) => {
	return <Form {...restProps}>{loading ? <ImSpinner8 /> : children}</Form>;
};

export default Button;
