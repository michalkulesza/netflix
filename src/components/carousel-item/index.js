import React from "react";
import { Item, Container, Wrapper, Loader, ExpandedSmall } from "./styles/carousel-item";

const CarouselItem = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

CarouselItem.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

CarouselItem.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

CarouselItem.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

CarouselItem.ExpandedSmall = ({ children, ...restProps }) => {
	return <ExpandedSmall {...restProps}>{children}</ExpandedSmall>;
};

export default CarouselItem;
