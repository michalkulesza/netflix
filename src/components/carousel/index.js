import React from "react";
import {
	Main,
	Category,
	Overlay,
	Button,
	Container,
	Wrapper,
	ItemsContainer,
	ItemWrapper,
	Item,
	Loader,
} from "./styles/carousel";

const Carousel = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Carousel.Category = ({ children, ...restProps }) => {
	return <Category {...restProps}>{children}</Category>;
};

Carousel.Overlay = ({ children, ...restProps }) => {
	return <Overlay {...restProps}>{children}</Overlay>;
};

Carousel.Button = ({ children, ...restProps }) => {
	return <Button {...restProps}>{children}</Button>;
};

Carousel.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

Carousel.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

Carousel.ItemsContainer = ({ children, ...restProps }) => {
	return <ItemsContainer {...restProps}>{children}</ItemsContainer>;
};

Carousel.ItemWrapper = ({ children, ...restProps }) => {
	return <ItemWrapper {...restProps}>{children}</ItemWrapper>;
};

Carousel.Item = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

Carousel.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

export default Carousel;
