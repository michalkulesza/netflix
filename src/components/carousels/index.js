import React from "react";
import { Main } from "./styles/carousels";

const Carousels = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

export default Carousels;
