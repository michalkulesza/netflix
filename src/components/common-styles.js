import { keyframes } from "styled-components/macro";

export const LoadingAnimation = keyframes`
	from {
		background-position-x: 0vw;
	}
	to {
		background-position-x: 100vw;
	}
`;
