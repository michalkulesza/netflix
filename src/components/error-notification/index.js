import React from "react";
import { Main, Wrapper } from "./styles/error-notification";

const ErrorNotification = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

ErrorNotification.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

export default ErrorNotification;
