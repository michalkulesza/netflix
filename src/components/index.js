export { default as Header } from "./header";
export { default as Navbar } from "./navbar";
export { default as Jumbotron } from "./jumbotron";
export { default as Faq } from "./faq";
export { default as Newsletter } from "./newsletter";
export { default as Footer } from "./footer";
export { default as Button } from "./button";
export { default as Form } from "./form";
export { default as Carousel } from "./carousel";
export { default as Carousels } from "./carousels";
export { default as Item } from "./item";
export { default as ItemExpanded } from "./item-expanded";
export { default as Details } from "./item-details/details";

export { default as ItemDetailsHeader } from "./item-details-header";
export { default as ItemDetailsInfo } from "./item-details-info";
export { ItemDetailsEpisodes } from "./item-details-episodes";
export { ItemDetailsEpisode } from "./item-details-episodes";
