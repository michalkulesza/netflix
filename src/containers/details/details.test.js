import React from "react";
import { mount } from "enzyme";
import DetailsContainer from "./details";
import * as types from "../../redux/types";
import configureStore from "redux-mock-store";
import * as ReactRedux from "react-redux";
import { Details } from "../../components";

let useEffect;
let useDispatch;
let store = configureStore()({});

const mockUseEffect = () => {
	useEffect.mockImplementationOnce(f => f());
};

beforeEach(() => {
	useEffect = jest.spyOn(React, "useEffect");
	useDispatch = jest.spyOn(ReactRedux, "useDispatch");
	mockUseEffect();
	mockUseEffect();
	store.clearActions();
});

describe("<DetailsContainer/> with no state", () => {
	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<DetailsContainer />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should return null", () => {
		expect(wrapper.isEmptyRender()).toEqual(true);
	});
});

describe("<DetailsContainer/> with state", () => {
	store = configureStore()({
		toggles: {
			globalMute: false,
		},
		user: {
			liked: [],
			disliked: [],
			list: [],
		},
		fetchEpisodes: {
			data: [],
			isUpdating: false,
		},
		toggles: {
			isDetails: true,
		},
		misc: {
			detailsPosition: {},
			headerVideo: {
				src: "/video",
				logo: "/logo",
			},
		},
		fetchDetails: {
			details: {
				name: "testName",
				first_air_date: "2020-10-20",
				number_of_seasons: 8,
				overview: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Et, perspiciatis?",
				genres: [{ name: "testGenre" }, { name: "testGenre" }],
				created_by: [{ name: "testCreator" }],
			},
			cast: [{ name: "Test" }, { name: "Test" }, { name: "Test" }],
			ageRestriction: 18,
		},
	});

	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<DetailsContainer />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should not return null", () => {
		expect(wrapper.isEmptyRender()).toEqual(false);
	});

	it("should render details", () => {
		expect(wrapper.find(Details.AboutHeader)).toHaveLength(1);
		expect(wrapper.find(Details.AboutPiece)).toHaveLength(4);
	});

	it("should fire handleClose on overlay click", () => {
		const expectedAction = [{ payload: false, type: types.SET_IS_DETAILS }];

		wrapper.find('[data-testid="overlay"]').hostNodes().simulate("mouseDown");
		expect(store.getActions()).toEqual(expectedAction);
	});
});
