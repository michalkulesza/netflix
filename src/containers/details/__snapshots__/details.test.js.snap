// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsContainer/> with no state should match the snapshot 1`] = `""`;

exports[`<DetailsContainer/> with state should match the snapshot 1`] = `"<div class=\\"details__Container-k62cua-0 fVwPdr\\"><div data-testid=\\"overlay\\" class=\\"details__OverlayTrigger-k62cua-1 fXcExH\\"></div><div class=\\"details__Main-k62cua-2 bZfRwe\\"><div class=\\"details-header__Header-dmikil-0 dYAEVp\\"><video type=\\"video/mp4\\" src=\\"/video\\" class=\\"details-header__Video-dmikil-1 gixNNu\\"></video><img src=\\"placeholder_w.jpg\\" class=\\"details-header__Cover-dmikil-2 VXGPf\\"><div class=\\"details-header__Overlay-dmikil-3 AIpJn\\"><div class=\\"details-header__OverlayHalf-dmikil-4 xFnRs\\"><img src=\\"/logo\\" alt=\\"Video Logo\\" class=\\"details-header__Logo-dmikil-5 KAkvI\\"><div class=\\"details-header__ButtonsContainer-dmikil-6 kIMfuj\\"><button data-testid=\\"playButton\\" class=\\"button__Square-nas94j-1 gzHUah\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M3,22.0000002 L21,12 L3,2 L3,22.0000002 Z M5,19 L17.5999998,11.9999999 L5,5 L5,19 Z M7,16 L14.1999999,12 L7,8 L7,16 Z M9,13 L10.8,12 L9,11 L9,13 Z\\"></path></svg> Play</button><div class=\\"button__Container-nas94j-3 kdslFq\\"><button data-testid=\\"listButton\\" class=\\"button__Round-nas94j-4 gGTyNM\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M19 11L13 11 13 5 11 5 11 11 5 11 5 13 11 13 11 19 13 19 13 13 19 13z\\"></path></svg></button><div class=\\"button__Label-nas94j-2 fApCpL\\">Add to My List</div></div><div class=\\"button__Container-nas94j-3 kdslFq\\"><button data-testid=\\"likeButton\\" class=\\"button__Round-nas94j-4 gGTyNM\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M20,8h-5.612l1.123-3.367c0.202-0.608,0.1-1.282-0.275-1.802S14.253,2,13.612,2H12c-0.297,0-0.578,0.132-0.769,0.36 L6.531,8H4c-1.103,0-2,0.897-2,2v9c0,1.103,0.897,2,2,2h3h10.307c0.829,0,1.581-0.521,1.873-1.298l2.757-7.351 C21.979,12.239,22,12.12,22,12v-2C22,8.897,21.103,8,20,8z M4,10h2v9H4V10z M20,11.819L17.307,19H8V9.362L12.468,4l1.146,0 l-1.562,4.683c-0.103,0.305-0.051,0.64,0.137,0.901C12.377,9.846,12.679,10,13,10h7V11.819z\\"></path></svg></button><div class=\\"button__Label-nas94j-2 fApCpL\\">I like this</div></div><div class=\\"button__Container-nas94j-3 kdslFq\\"><button data-testid=\\"dislikeButton\\" class=\\"button__Round-nas94j-4 gGTyNM\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M20,3h-3H6.693C5.864,3,5.112,3.521,4.82,4.298l-2.757,7.351C2.021,11.761,2,11.88,2,12v2c0,1.103,0.897,2,2,2h5.612 L8.49,19.367c-0.203,0.608-0.101,1.282,0.274,1.802C9.14,21.689,9.746,22,10.388,22H12c0.297,0,0.578-0.132,0.769-0.36l4.7-5.64 H20c1.103,0,2-0.897,2-2V5C22,3.897,21.103,3,20,3z M11.531,20h-1.145l1.562-4.684c0.103-0.305,0.051-0.64-0.137-0.901 C11.623,14.154,11.321,14,11,14H4v-1.819L6.693,5H16v9.638L11.531,20z M18,14V5h2l0.001,9H18z\\"></path></svg></button><div class=\\"button__Label-nas94j-2 fApCpL\\">Not for me</div></div></div></div><div class=\\"details-header__OverlayHalf-dmikil-4 xFnRs\\"><div class=\\"button__Container-nas94j-3 kdslFq\\"><button data-testid=\\"closeButton\\" class=\\"button__Round-nas94j-4 icIDoW\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M3,3 L21,21 M3,21 L21,3\\"></path></svg></button></div><div class=\\"button__Container-nas94j-3 kdslFq\\"><button data-testid=\\"muteButton\\" class=\\"button__Round-nas94j-4 gGTyNM\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 512 512\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M275.5 96l-96 96h-96v128h96l96 96V96zm51.46 27.668l-4.66 17.387c52.066 13.95 88.2 61.04 88.2 114.945 0 53.904-36.134 100.994-88.2 114.945l4.66 17.387C386.81 372.295 428.5 317.962 428.5 256c0-61.963-41.69-116.295-101.54-132.332zm-12.425 46.365l-4.658 17.387C340.96 195.748 362.5 223.822 362.5 256s-21.54 60.252-52.623 68.58l4.658 17.387C353.402 331.552 380.5 296.237 380.5 256c0-40.238-27.098-75.552-65.965-85.967zm-12.424 46.363l-4.657 17.387C307.55 236.49 314.5 245.547 314.5 256s-6.95 19.51-17.047 22.217l4.658 17.387c17.884-4.792 30.39-21.09 30.39-39.604 0-18.513-12.506-34.812-30.39-39.604z\\"></path></svg></button></div></div></div></div><div class=\\"details-info__Main-y9dbgt-0 hqPYPP\\"><div class=\\"details-info__Half-y9dbgt-1 gkhROt\\"><div class=\\"details-info__Content-y9dbgt-2 dOFEgX\\"><p>96% Match</p> 2020 <span>18</span> 8 Seasons</div><div class=\\"details-info__Description-y9dbgt-3 cRzHom\\">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Et, perspiciatis?</div></div><div class=\\"details-info__Half-y9dbgt-1 gkhROt\\"><div class=\\"details-info__Cast-y9dbgt-4 iYvcrP\\"><span>Cast:</span> Test, Test, Test</div><div class=\\"details-info__Cast-y9dbgt-4 iYvcrP\\"><span>Genres:</span> testGenre, testGenre</div></div></div><div class=\\"details-episodes__Episodes-q2f4d2-0 kNwKbG\\"><div class=\\"details-episodes__EpisodesHeader-q2f4d2-1 hrxrGP\\">Episodes<div class=\\"details-episodes__EpisodesSeasons-q2f4d2-2 gwFCAB\\"><button data-testid=\\"SeasonsButton\\" class=\\"details-episodes__EpisodesSeasonsButton-q2f4d2-3 dcCfkk\\"><span>Season 1</span><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M11.998 17L18.998 9 4.998 9z\\"></path></svg></button><ul class=\\"details-episodes__EpisodesSeasonsList-q2f4d2-4 eJeDmD\\"><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 1</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 2</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 3</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 4</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 5</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 6</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 7</li><li class=\\"details-episodes__EpisodesSeason-q2f4d2-5 fQVBUV\\">Season 8</li></ul></div></div> <div class=\\"details-episodes__EpisodesList-q2f4d2-6\\"></div></div><div class=\\"details-related__Related-sc37ur-0 junUEP\\"><div class=\\"details-related__RelatedHeader-sc37ur-1 gfVNED\\">More Like This</div><div class=\\"details-related__RelatedItems-sc37ur-2 dAqrBA\\"></div></div><div class=\\"details__About-k62cua-3 dusmSz\\"><div class=\\"details__AboutHeader-k62cua-4 ebogFE\\">About testName</div><div class=\\"details__AboutPiece-k62cua-5 cltGgZ\\">Creators:<p>testCreator</p></div><div class=\\"details__AboutPiece-k62cua-5 cltGgZ\\">Cast:<p>Test, Test, Test</p></div><div class=\\"details__AboutPiece-k62cua-5 cltGgZ\\">Genres:<p>testGenre, testGenre</p></div><div class=\\"details__AboutPiece-k62cua-5 cltGgZ\\">Maturity rating <span>18</span><p>Suitable for ages 18 and up</p></div></div></div></div>"`;
